crates/nargo_cli/src/cli/fs/common_reference_string.rs

```diff
@@ -35,15 +36,20 @@ pub(crate) fn update_common_reference_string<B: CommonReferenceString>(
 
     let runtime = Builder::new_current_thread().enable_all().build().unwrap();
 
-    // TODO(#1391): Implement retries
     // If the read data is empty, we don't have a CRS and need to generate one
-    let fut = if common_reference_string.is_empty() {
-        backend.generate_common_reference_string(circuit)
-    } else {
-        backend.update_common_reference_string(common_reference_string.to_vec(), circuit)
-    };
-
-    runtime.block_on(fut)
+    for i in 0..MAX_CRS_DOWNLOAD_RETRIES {
+        let fut = if common_reference_string.is_empty() {
+            backend.generate_common_reference_string(circuit)
+        } else {
+            backend.update_common_reference_string(common_reference_string.to_vec(), circuit)
+        };
+
+        let res = runtime.block_on(fut);
+        if res.is_ok() || i == 4 {
+            return res;
+        }
+    }
+    unreachable!();
```

Comment:
```suggestion
   Err(SomeErrorHere)
```

Reply:
can you elaborate on how would I return the error?
I get this 
```mismatched types
expected associated type `<B as CommonReferenceString>::Error`
            found struct `MyError`
            ``` 
            
every time I try to send an error back. even though my error implements `std::error::Error` similar to `CommonReferenceString`